WordPress/Classes/ViewRelated/Post/PostMediaUploadViewModel.swift

```diff
@@ -0,0 +1,168 @@
+        MediaImageService.isThubmnailSupported(for: media.mediaType) ? 40 : 24
+    }
+
+    enum State {
+        case uploaded
+        case uploading
+    }
+
+    deinit {
+        retryTimer?.invalidate()
+    }
+
+    init(media: Media, coordinator: MediaCoordinator) {
+        self.media = media
+        self.coordinator = coordinator
+
+        title = media.filename ?? "–"
+        update()
+
+        NotificationCenter.default.addObserver(self, selector: #selector(didUpdateReachability), name: .reachabilityChanged, object: nil)
+    }
+
+    fileprivate func update() {
+        self.state = media.isUploadNeeded ? .uploading : .uploaded
+        self.fileSize = media.filesize?.int64Value ?? 0 // Should never be `0`
+
+        if media.remoteStatus == .failed, retryTimer == nil {
+            retryTimer = Timer.scheduledTimer(withTimeInterval: nextRetryDelay, repeats: false) { [weak self] _ in self?.retry() }
+        }
+
+        if media.isUploadNeeded {
+            if let progress = coordinator.progress(for: media),
+               let uploadProgress = progress.userInfo[.uploadProgress] as? Progress {
+                let completed = Int64(Double(fileSize) * uploadProgress.fractionCompleted)
+                self.fractionCompleted = uploadProgress.fractionCompleted
+                self.completed = max(completed, self.completed)
+            }
+            details = Strings.progress(completed: completed, total: fileSize)
+        } else {
+            fractionCompleted = 1.0
+            details = ByteCountFormatter.string(fromByteCount: fileSize, countStyle: .file)
+        }
+    }
+
+    private func retry() {
+        retryTimer = nil
+        coordinator.retryMedia(media)
+    }
+
+    @objc private func didUpdateReachability(_ notification: Foundation.Notification) {
+        guard let reachable = notification.userInfo?[Foundation.Notification.reachabilityKey],
+              (reachable as? Bool) == true else {
+            return
+        }
+        if (media.error as? URLError)?.code == .notConnectedToInternet {
+            retry()
+        }
+    }
+
+    @MainActor
+    func loadThumbnail() async {
+        do {
+            if MediaImageService.isThubmnailSupported(for: media.mediaType) {
+                thumbnail = try await MediaImageService.shared.image(for: media, size: .small)
+            } else {
+                thumbnail = SiteMediaDocumentInfoViewModel.make(with: media).image
+            }
+        } catch {
+            // Continue showing placeholder
+        }
+    }
+}
+
+private extension Media {
+    var isUploadNeeded: Bool {
+        mediaID == nil || mediaID?.intValue == 0
+    }
+}
+
+private enum Strings {
+    static func progress(completed: Int64, total: Int64) -> String {
+        let format = NSLocalizedString("postMediaUploadStatusView.progress", value: "%@ of %@", comment: "Shows the upload progress with two parameters: preformatted completed and total bytes")
```

Comment:
Let's use positional placeholders (ref: pbMoDN-1Df-p2) 
```suggestion
        let format = NSLocalizedString("postMediaUploadStatusView.progress", value: "%1$@ of %2$@", comment: "Shows the upload progress with two preformatted parameters: %1$@ is the placeholder for completed bytes, and %2$@ is the placeholder for total bytes")
```